rust/azure_iot_operations_mqtt/src/session.rs

```diff
@@ -705,27 +727,34 @@ impl SessionExitHandle {
             // disconnect handle from the weak reference, meaning the Session (and the client
             // inside it has not dropped
             .or(Err(SessionExitError {
-                attempted: false,
                 kind: SessionExitErrorKind::Detached,
             }))
     }
 
     /// Forcefully end the MQTT session running in the [`Session`] that created this handle.
     /// This will cause the [`Session::run()`] method to return.
     ///
-    /// The [`Session`] will be granted a period of 1 second to attempt a graceful exit before
-    /// forcing the exit. If the exit is forced, the broker will not be aware the MQTT session
-    /// has ended.
+    /// The [`Session`] will attempt a graceful exit before forcing the exit.
+    /// If the exit is forced, the server will not be aware the MQTT session has ended.
     ///
     /// Returns true if the exit was graceful, and false if the exit was forced.
     #[allow(clippy::must_use_candidate)]
     pub fn force_exit(&self) -> bool {
-        if self.try_exit().is_ok() {
-            true
-        } else {
-            log::debug!("Session not connected, forcing exit immediately");
-            self.force_exit.notify_one();
-            false
+        log::debug!("Attempting to exit session gracefully");
+        match self.try_exit() {
+            Ok(()) => {
+                log::debug!("Session exited gracefully");
+                true
+            }
+            Err(e) => {
+                if e.kind() == SessionExitErrorKind::Detached {
+                    log::debug!("Session already detached, no action needed");
```

Comment:
this case is safe to not notify the force_exit because this would only be the case if the task that this cleans up is already cleaned up, right?